### PR TITLE
Add dynlink to builtin libs

### LINKED ISSUE #1264: dynlink missing in meta.ml
`dune` fails to find `dynlink` in the absence of `ocamlfind`.

### PR DESCRIPTION
**Pull request by rgrinberg**
Fix #1264

### CODE REVIEW

**Review by emillon — APPROVED** [review]

File: src/meta.ml [hunk 1]
```diff
@@ -257,14 +257,23 @@ let builtins ~stdlib_dir =
         ]
     }
   in
+  let dynlink =
```
Comment (nojb) [inline src/meta.ml]: Shouldn't this be just `let dynlink = simple "dynlink" ~dir:"+" []` like e.g. `bigarray` above ?

Comment (rgrinberg) [reply src/meta.ml]: Indeed. Thanks.

**Comment by rgrinberg** [comment]
Btw, does bigarray still depend on Unix? This looks a bit suspicious to me:

```
  let bigarray = simple "bigarray" ["unix"] ~dir:"+" in
```

**Comment by dra27** [comment]
LGTM. The bigarray library itself does still depend on Unix IIRC - that's the "legacy" version of it with memory mapping, as opposed to the standard library version.

**Review by dra27 — APPROVED** [review]

**Comment by rgrinberg** [comment]
@nojb could you confirm that this fixes things for you

**Review by nojb — APPROVED** [review]

**Comment by nojb** [comment]
I have this patch on my local copy of `dune` and it works.

**Pull request merged**